src/annotator/util/test/buckets-test.js

```diff
@@ -1,44 +1,65 @@
-import { findClosestOffscreenAnchor, anchorBuckets } from '../buckets';
+import {
+  findClosestOffscreenAnchor,
+  computeAnchorPositions,
+  computeBuckets,
+} from '../buckets';
 import { $imports } from '../buckets';
 
-let tagIndex = 0;
-function fakeAnchorFactory(
-  offsetStart = 1,
-  offsetIncrement = 100,
-  boxHeight = 50
-) {
-  let highlightIndex = offsetStart;
-  return () => {
-    // In a normal `Anchor` object, `highlights` would be an array of
-    // DOM elements. Here, `highlights[0]` is the vertical offset (top) of the
-    // fake anchor's highlight box and `highlights[1]` is the height of the
-    // box. This is in used in conjunction with the mock for
-    // `getBoundingClientRect`, below
-    const anchor = {
-      annotation: { $tag: `t${tagIndex++}` },
-      highlights: [highlightIndex, boxHeight],
-    };
-    highlightIndex = highlightIndex + offsetIncrement;
-    return anchor;
-  };
-}
-
 describe('annotator/util/buckets', () => {
-  let fakeGetBoundingClientRect;
-
   let fakeAnchors;
+  let fakeAnchorPositions;
+  let fakeGetBoundingClientRect;
   let stubbedInnerHeight;
 
   beforeEach(() => {
-    const fakeAnchor = fakeAnchorFactory();
     fakeAnchors = [
-      fakeAnchor(), // top: 1, bottom: 51 — above screen
-      fakeAnchor(), // top: 101, bottom: 151 — above screen
-      fakeAnchor(), // top: 201, bottom: 251 — on screen
-      fakeAnchor(), // top: 301, bottom: 351 — on screen
-      fakeAnchor(), // top: 401, bottom: 451 — below screen
-      fakeAnchor(), // top: 501, bottom: 551 - below screen
+      // top: 1, bottom: 51 — above screen
+      { annotation: { $tag: 't0' }, highlights: [1, 50] },
+      // top: 101, bottom: 151 — above screen
+      { annotation: { $tag: 't1' }, highlights: [101, 50] },
+      // top: 201, bottom: 251 — on screen
+      { annotation: { $tag: 't2' }, highlights: [201, 50] },
+      // top: 301, bottom: 351 — on screen
+      { annotation: { $tag: 't3' }, highlights: [301, 50] },
+      // top: 401, bottom: 451 — below screen
+      { annotation: { $tag: 't4' }, highlights: [401, 50] },
+      // top: 501, bottom: 551 - below screen
+      { annotation: { $tag: 't5' }, highlights: [501, 50] },
+    ];
```

Comment:
All of these test cases currently has only a single "highlight". That's not too much of an issue since the whole collection is passed to `getBoundingClientRect` which returns the combined bounding box.

Reply:
I assumed that testing multiple (overlapping) highlights was done in `highlighter-test.js`

Reply:
`highlighter-test.js` does indeed test that `getBoundingClientRect` returns the union if a collection is passed in. There would be value in having an integration test somewhere that includes the case of an annotation with multiple highlights. That's out of scope for this PR.